terraform/modules/quorum-cluster-region/quorum-cluster-region-backup-event.tf

```diff
@@ -0,0 +1,399 @@
+        "logs:*"
+      ],
+      "Resource": "arn:aws:logs:*:*:*",
+      "Effect": "Allow"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "BackupLambda2" {
+   count = "${signum(lookup(var.maker_node_counts, var.aws_region, 0))}"
+   role = "${aws_iam_role.iam_for_BackupLambda.name}"
+   policy_arn = "${aws_iam_policy.BackupLambda_logging.arn}"
+}
+
+# Zip the Backup Lambda
+resource "null_resource" "BackupLambda" {
+  count = "${signum(lookup(var.maker_node_counts, var.aws_region, 0))}"
+  triggers {
+    main = "${sha256(file("${var.BackupLambda_source_file}"))}"
+  }
+  # destroy
+  provisioner "local-exec" {
+    when = "destroy"
+    command = "rm ${var.aws_region}-${var.BackupLambda_output_path}"
+    on_failure = "continue"
+  }
+  // # creat
+  // provisioner "local-exec" {
+  //   command = "zip -j ${var.aws_region}-${var.BackupLambda_output_path} ${var.BackupLambda_source_file}"
+  // }
+  # the creation is in "archive_file" "BackupLambda"
+}
+
+data "archive_file" "BackupLambda" {
+  count = "${signum(lookup(var.maker_node_counts, var.aws_region, 0))}"
+  type        = "zip"
+  source_file  = "${var.BackupLambda_source_file}"
+  output_path = "${var.aws_region}-${var.BackupLambda_output_path}"
+  depends_on = ["null_resource.BackupLambda"]
+}
+
+resource "aws_kms_key" "a" {
+  count = "${signum(lookup(var.maker_node_counts, var.aws_region, 0))}"
+  description = "Used for encrypting SSH keys on S3"
+  tags {
+     name = "BackupLambda-${var.network_id}-${var.aws_region}-KMS"
+  }
+}
+
+output "aws_kms_key_test" {
+  value = "${aws_kms_key.a.*.id}"
+}
+
+# Encrypt the contents of the file located at var.private_key_path
+data "aws_kms_ciphertext" "EncryptSSHKey" {
+  count = "${signum(lookup(var.maker_node_counts, var.aws_region, 0))}"
+  key_id = "${aws_kms_key.a.id}"  
+  plaintext = "${file("${var.private_key_path}")}"
+}
+
+# Save the encrypted contents to the file specified at filename
+resource "local_file" "EncryptedSSHKey" {
+  count = "${signum(lookup(var.maker_node_counts, var.aws_region, 0))}"
+  content = "${base64decode("${data.aws_kms_ciphertext.EncryptSSHKey.ciphertext_blob}")}"
+  filename = "${var.enc_ssh_path}-${var.aws_region}"
+}
+
+resource "aws_kms_grant" "a" {
+  count = "${signum(lookup(var.maker_node_counts, var.aws_region, 0))}"
+  name              = "kms-grant-${var.network_id}-${var.aws_region}"
+  key_id            = "${aws_kms_key.a.key_id}"
+  grantee_principal = "${aws_iam_role.iam_for_BackupLambda.arn}"
+  operations        = ["Encrypt", "Decrypt"]
+}
+
+output "aws_s3_bucket" {
+  value = "${element(concat(aws_s3_bucket.quorum_backup.*.id, list("")), 0)}"
+}
+
+resource "aws_security_group" "allow_all" {
+    count = "${signum(lookup(var.maker_node_counts, var.aws_region, 0))}"
+    name        = "BackupLambdaSSH-${var.network_id}-${var.aws_region}-allow_all"
+    description = "Allow all outgoing traffic"
+    vpc_id = "${aws_vpc.quorum_cluster.id}"
+
+    egress {
```

Comment:
I prefer to use `aws_security_group_rule` resources rather than specifying rules inline